rflx/model/model.py

```diff
@@ -59,8 +58,13 @@ def write_specification_files(self, output_dir: Path) -> None:
         for package, specification in self.create_specifications().items():
             (output_dir / f"{package.flat.lower()}.rflx").write_text(specification)
 
-    def _validate(self) -> None:
+    def _add_missing_types_and_validate(self) -> None:
         error = self._check_duplicates()
+
+        _types = [t for s in self._sessions for t in s.non_private_types.values()] + [*self._types]
+        _types = [d for t in _types for d in t.dependencies_postorder]
+        self._types = list(unique(_types))
+
         error += self._check_conflicts()
         error.propagate()
```

Comment:
Local variables should not have a leading underscore.

Reply:
Done.